isis/tests/ColumnTests.cpp

```diff
@@ -0,0 +1,232 @@
+  Column column;
+
+  EXPECT_DOUBLE_EQ(column.Precision(), 4);
+  EXPECT_DOUBLE_EQ(column.Width(), 0);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, column.Name(), QString(""));
+  EXPECT_EQ(column.Alignment(), Column::NoAlign);
+  EXPECT_EQ(column.DataType(), Column::NoType);
+}
+
+//Tests that the initialization constructor works as intended
+TEST(Column, InitConstructor) {
+  Column column("col1", 25, Column::Pixel, Column::Left);
+
+  EXPECT_DOUBLE_EQ(column.Precision(), 4);
+  EXPECT_DOUBLE_EQ(column.Width(), 25);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, column.Name(), QString("col1"));
+  EXPECT_EQ(column.Alignment(), Column::Left);
+  EXPECT_EQ(column.DataType(), Column::Pixel);
+}
+
+//Tests SetName function
+TEST(Column, SetName) {
+  Column column;
+  column.SetName("Test Column");
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, column.Name(), QString("Test Column"));
+}
+
+//Tests SetWidth function
+TEST(Column, SetWidth) {
+  Column column;
+  column.SetWidth(100);
+
+  EXPECT_DOUBLE_EQ(column.Width(), 100);
+}
+
+//Tests SetType function with every member of the Type enum
+TEST_P(Type, SetType) {
+  Column column;
+  column.SetType(GetParam());
+
+  EXPECT_EQ(column.DataType(), GetParam());
+}
+
+INSTANTIATE_TEST_CASE_P(Column, Type, ::testing::Values(
+  Column::NoType, Column::Integer, Column::Real, Column::String, Column::Pixel));
+
+//Tests SetAlignment function with every member of the Align enum
+TEST_P(Align, SetAlignment) {
+  Column column;
+  column.SetAlignment(GetParam());
+
+  EXPECT_EQ(column.Alignment(), GetParam());
+}
+
+INSTANTIATE_TEST_CASE_P(Column, Align, ::testing::Values(
+  Column::NoAlign, Column::Right, Column::Left, Column::Decimal));
+
+//Tests SetPrecision function with Real type and Pixel type.
+//These are the only two types expected to work with SetPrecision
+TEST(Column, SetPrecision) {
+  Column column;
+  column.SetType(Column::Real);
+  column.SetPrecision(10);
+  EXPECT_DOUBLE_EQ(column.Precision(), 10);
+
+  column.SetType(Column::Pixel);
+  column.SetPrecision(15);
+  EXPECT_DOUBLE_EQ(column.Precision(), 15);
+}
+
+//Tests Name getter function
+TEST(Column, Name) {
+  Column column;
+  column.SetName("Test Column");
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, column.Name(), QString("Test Column"));
+}
+
+//Tests width getter function
+TEST(Column, Width) {
+  Column column;
+  column.SetWidth(15);
+
+  EXPECT_DOUBLE_EQ(column.Width(), 15);
+}
+
+//Tests DataType getter function
+TEST(Column, DataType) {
+  Column column;
+  column.SetType(Column::Pixel);
+
+  EXPECT_EQ(column.DataType(), Column::Pixel);
+}
+
+//Tests Alignment getter function
+TEST(Column, Alignment) {
+  Column column;
+  column.SetAlignment(Column::Left);
+
+  EXPECT_EQ(column.Alignment(), Column::Left);
+}
+
+//Tests Precision getter function
+TEST(Column, Precision) {
+  Column column;
+  column.SetType(Column::Real);
+  column.SetPrecision(100);
+
+  EXPECT_DOUBLE_EQ(column.Precision(), 100);
+}
```

Comment:
All of the accessor tests are doing the same thing as the mutator tests. You can get rid of all of the accessor tests.